EIPS/eip-7743.md

```diff
@@ -0,0 +1,120 @@
+
+Connecting OIDC identity and smart accounts has been a very interesting but challenging problem. The most common ways of tackling the issue is by utilizing Multi-Party Computation(MPC). However, the limitation of the MPC solution is obvious:
+
+1. **Centralization**: It relies on a third-party service to sign and aggregate the signature which introduces centralization risk such as single point of failure and vendor lock-in.
+2. **Privacy**: the separation between the users' Web2 identity to their Web3 address cannot be cryptographically guaranteed and results in higher cost of trust between the users and the wallet service providers.
+
+In this EIP, we propose a new model with OIDC ZK verification to address these concerns. The new model has the following characteristics:
+
+1. **ZK Verified**. Privacy will be guaranteed through ZK verification as the connection between Web2 identity and the Web3 account will be hidden.
+2. **Decentralized**. The ZK proof generation can be done on client side without involving any third-party service. The ZK-proof aggregation is also stateless, so it can be done via any gateway without introducing extra centralization dependencies.
+3. **Affordable**. Daily operations could be signed and executed with the local smart account operator. The ZK verification will only be used to reset the account if the operator key is lost or stolen to ensure the whole setup is affordable. ZK proofs aggregation has also proven to be viable and paves the way for cheaper verification cost at scale.
+
+## Specification
+
+The keywords “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+### Definitions
+
+**Account**: The user's smart account contract
+
+**Operator**: The signer to sign and exectue transactions. This can be reset by the verifier if lost or stolen.
+
+**Identity Provider(IDP)**: The service to authenticate users and provide signed ID token
+
+**Client**: The application to authenticate users and generate the ZK proof
+
+**Aggregrator**: The offchain service to aggregate ZK proof from multiple users
+
+**Verifier**: The on-chain contract to verify the ZK proof and recover account by resetting the operator
+
+### Example Flow
+
+![The example workflow](../assets/eip-7743/workflow.png)
+
+### Interface
+
+```
+interface IOpenIdVerifier {
+    struct Action {
+        address account;
+        bytes data;
+    }
+ 
+    // @notice get verification key of the open id authenticator
+    function getVerificationKey() external view returns(bytes memory);
+ 
+    // @notice get id hash of account
+    function getIdHash(address account) external view returns(bytes32);
+
+    // @notice the function verifies the proof and executes the action if it's valid
+    // @params action: the action to execute
+    //         proof: data to prove the account ownership
+    function verifyAndCall(
```

Comment:
This could be used beyond ZKP use case but more general. Meanwhile, regular permission mechanism could also be used to approve things with ZKP proofs. Help us understand better what ZKP+AA makes the requirement of the ERC interface special?

Reply:
The design is intentional. Regular permission mechanism per account is hard to deal with aggregated ZK proof, in which case we will need a third party contract(the verifier in this case) to process multiple accounts' recovery in batch. Another option is to merge it with the ERC4337 Aggregator in which case we will not need verifyAndCall but just verify function and the verifier has to be compatible with the IAggregator interface at ERC4337